compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -104,7 +104,7 @@ pub struct Diagnostic {
     pub(crate) level: Level,
 
     pub messages: Vec<(DiagnosticMessage, Style)>,
-    pub code: Option<DiagnosticId>,
+    pub code: Option<String>,
```

Comment:
Is there a way to store an integer instead of a string here? An `ErrorCode` type?

Reply:
Yes! You're going to love #119972, which I just finished :)